app/models/user.rb

```diff
@@ -2,9 +2,39 @@ class User < ApplicationRecord
   enum role: {normal: 0, admin: 1, manager: 2}
   
   devise :database_authenticatable, :registerable,
-    :recoverable, :rememberable, :validatable
+    :recoverable, :rememberable, :validatable,
+    :omniauthable, omniauth_providers: [:facebook]
 
-  has_many :stores
-  has_many :comments
-  has_many :bills
+  has_many :stores, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :bills, dependent: :destroy
+
+  validates :email, format: {with: Settings.user.VALID_EMAIL_REGEX},
+    presence: true, length: {maximum: Settings.user.max_email_length},
+    uniqueness: {case_sensitive: false}
+  validates :password, presence: true, length:
+    {minimum: Settings.user.min_password_length}, allow_nil: true
+  validates :name, presence: true, length:
+    {minimum: Settings.user.min_name_length, maximum: Settings.user.max_name_length}
+
+  class << self
+    def new_with_session params, session
+      super.tap do |user|
```

Comment:
toàn bị thừa super thế :v

Reply:
Tại cái này em checkout từ nhánh trước khi anh check nên vẫn còn a.